### DISCUSSION TITLE
Conditional for each trial

### DISCUSSION THREAD

**Discussion by FelicityF** [jspsych/jsPsych#1750] [Q&A]
I ran a study last semester in which participants either received no feedback, negative feedback, or positive feedback. However, this time I want to expose participants to each type of feedback. So, my idea was to create a time line for the feedback trial and based on a randomly drawn number each item would have the type of feedback assigned to it: 

      var Feedback = [];

      //Feedback
      var Feedback_NO = [fixation, trial, feedbackNO];
      var Feedback_NG = [fixation, trial, feedbackNEG];
      var Feedback_P = [fixation, trial, feedbackP];

      var conditional = Math.floor(Math.random() * 3) + 1;

              if(conditional == 1){
              		Feedback = timeline.concat(Feedback_NO);
              	}else if(conditional == 2){
              		Feedback = timeline.concat(Feedback_NG);
              	}else{
              		Feedback = timeline.concat(Feedback_P);
              	}

        var procedure_training1 = {
            timeline: Feedback,
            timeline_variables: block1,
            randomize_order: true
                }; 

However, when I do it this way it seems that all the items receive the same kind of feedback (so it might only be drawn once?). What I would like to happen is that this happens for each trial within block1 instead. Any suggestions how I could make this happen?

**Comment by becky-gilbert** [comment]
Hi @FelicityF, yes, the way you've written this means that the feedback trial type is only selected once (when the page first loads). There are probably a few ways to do this, and I'm not sure which is best for your situation. Here's one idea: would it be possible for you to create a single `feedback` trial (instead of 3 separate trials: `feedbackNO`, `feedbackNEG`, and `feedbackP`) and use information passed in from timeline variables to change the feedback trial as necessary, based on the feedback type (none/positive/negative)? I think it would look like this in the trial information that you pass to timeline variables:
```js
var block1 = [
  { ... other trial info ... , feedback_condition: 'positive', feedback_stimulus: 'yay!'},
  { ... other trial info ... , feedback_condition: 'negative', feedback_stimulus: 'boo!'},
  { ... other trial info ... , feedback_condition: 'none', feedback_stimulus: ''}
];
```
Then your feedback trial might look like this: 
```js
var feedback = {
  type: 'html-keyboard-response',
  stimulus: jsPsych.timelineVariable('feedback_stimulus'),
  choices: jsPsych.NO_KEYS,
  duration: 3000,
  data: {feedback_condition: jsPsych.timelineVariable('feedback_condition')
};
```
You could also add the feedback duration to your trial information in timeline variables, if you wanted to set the feedback duration to 0 for the 'none' conditon.

This way your trial procedure timeline would always be the same:
```js
var procedure_training1 = {
  timeline: [fixation, trial, feedback],
  timeline_variables: block1,
  randomize_order: true
}; 
```
Does that work for you?
